javaagent-tooling/javaagent-tooling-java9/src/main/java/io/opentelemetry/javaagent/tooling/Java9LambdaTransformer.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.tooling;
+
+import io.opentelemetry.javaagent.bootstrap.LambdaTransformer;
+import java.lang.instrument.ClassFileTransformer;
+import java.lang.instrument.IllegalClassFormatException;
+
+/** lambda transformer with java9 jpms module compatibility */
+public class Java9LambdaTransformer implements LambdaTransformer {
+
+  private final ClassFileTransformer delegate;
+
+  public Java9LambdaTransformer(ClassFileTransformer delegate) {
+    this.delegate = delegate;
+  }
+
+  @Override
+  public byte[] transform(String className, Class<?> targetClass, byte[] classfileBuffer)
+      throws IllegalClassFormatException {
+
+    // lambda instrumentation happens only when the lambda is defined, thus the classBeingRedefined
```

Comment:
I think this comment isn't entirely correct any more because `classBeingRedefined` isn't really used